sdk/src/account_info.rs

```diff
@@ -92,17 +90,25 @@ impl<'a> AccountInfo<'a> {
     }
 
     pub fn deserialize_data<T: serde::de::DeserializeOwned>(&self) -> Result<T, bincode::Error> {
-        bincode::deserialize(&self.m.borrow().data)
+        bincode::deserialize(&self.data)
     }
 
     pub fn serialize_data<T: serde::Serialize>(&mut self, state: &T) -> Result<(), bincode::Error> {
-        if bincode::serialized_size(state)? > self.m.borrow().data.len() as u64 {
+        if bincode::serialized_size(state)? > self.data.len() as u64 {
             return Err(Box::new(bincode::ErrorKind::SizeLimit));
         }
         bincode::serialize_into(&mut self.m.borrow_mut().data[..], state)
     }
 }
 
+impl<'a> Deref for AccountInfo<'a> {
+    type Target = AccountInfoMut<'a>;
+
+    fn deref(&self) -> &Self::Target {
+        unsafe { self.m.as_ptr().as_ref().unwrap() }
```

Comment:
@garious What do you think of this `unsafe` block?  I'm planning to also apply this to `account::KeyedAccount`

Reply:
That looks unsafe to me. Why do you think it's safe?

Reply:
The question is does this unsafeness unsafe for us?  There is no chance of panic and being we are single-threaded where it's used do we care if we read the value while its also held mutably elsewhere?

Reply:
`mut` means both "mutable" and "mutually exclusive access". It's not obvious to me that your use of `unsafe` is safe.  Also, if it were safe, I'd think RefCell would implement Deref itself, no?

Reply:
I can understand not having two mutable references.  This unsafe block results in a non-mutable reference.  So mutually exclusive mutable access is retained while allowing a non-mutuable access to occur at the same time.  Do you have a scenario in mind where this would be "unsafe" for us where this is used?

Reply:
My biggest concern would be of a compiler optimization that fires because it thinks it has all references.

Do you have a use case that's so valuable that we should risk the potential need to debug a seemingly random crash?

Reply:
The use case is not strong, mainly simplification of the accessor syntax.  I'll scrap this PR.  

I'm doubtful that this would cause a crash due to a compiler optimization because my understanding of `unsafe` means "Know what you are doing" rather than "You are heading into undefined language/compiler behavior".  But I can't say for sure :-)

Reply:
Some light weekend reading? https://doc.rust-lang.org/nomicon/index.html

Reply:
Yup, been through most of that :-)